src/DotLiquid.Tests/StandardFilterTests.cs

```diff
@@ -1572,5 +1603,80 @@ public void TestWhere_NonStringCompare()
 {% endfor %}",
                 localVariables: Hash.FromAnonymousObject(new { products }));
         }
+
+        [Test]
+        public void TestConcat()
+        {
+            var array1 = new String[] { "one", "two" };
+            var array2 = new String[] { "alpha", "bravo" };
+            var arrayCombined = new String[] { "one", "two", "alpha", "bravo" };
+
+            Assert.AreEqual(null, StandardFilters.Concat(null, null));
+            Assert.AreEqual(array1, StandardFilters.Concat(array1, null));
+            Assert.AreEqual(array1, StandardFilters.Concat(null, array1));
+            Assert.AreEqual(arrayCombined, StandardFilters.Concat(array1, array2));
+        }
+
+        [Test]
+        public void TestConcat_LiquidSample_SingleFilter()
+        {
+            Helper.AssertTemplateResult(
+                expected: "\r\n\r\n\r\n\r\n- apples\r\n\r\n- oranges\r\n\r\n- peaches\r\n\r\n- carrots\r\n\r\n- turnips\r\n\r\n- potatoes\r\n",
+                template: @"{% assign fruits = 'apples, oranges, peaches' | split: ', ' %}
+{% assign vegetables = 'carrots, turnips, potatoes' | split: ', ' %}
+{% assign everything = fruits | concat: vegetables %}
+{% for item in everything %}
+- {{ item }}
+{% endfor %}");
+        }
+
+        [Test]
+        public void TestConcat_LiquidSample_ChainedFilters()
+        {
+            Helper.AssertTemplateResult(
+                expected: "\r\n\r\n\r\n\r\n\r\n- apples\r\n\r\n- oranges\r\n\r\n- peaches\r\n\r\n- carrots\r\n\r\n- turnips\r\n\r\n- potatoes\r\n\r\n- chairs\r\n\r\n- tables\r\n\r\n- shelves\r\n",
+                template: @"{% assign fruits = 'apples, oranges, peaches' | split: ', ' %}
+{% assign vegetables = 'carrots, turnips, potatoes' | split: ', ' %}
+{% assign furniture = 'chairs, tables, shelves' | split: ', ' %}
+{% assign everything = fruits | concat: vegetables | concat: furniture %}
+{% for item in everything %}
+- {{ item }}
+{% endfor %}");
+        }
+
+        [Test]
+        public void TestReverse()
+        {
+            var array = new String[] { "one", "two", "three" };
+            var arrayReversed = new String[] { "three", "two", "one" };
+
+            Assert.AreEqual(null, StandardFilters.Reverse(null));
+            Assert.AreEqual(arrayReversed, StandardFilters.Reverse(array));
+            Assert.AreEqual(array, StandardFilters.Reverse(arrayReversed));

```

Comment:
Should IEnumerable return be tested by CollectionAssert.AreEqual?

Reply:
CollectionAssert.AreEqual